src/common/content_type.rs

```diff
@@ -148,6 +158,16 @@ mod tests {
         );
     }
 
+    #[test]
+    fn from_str() {
+        assert_eq!(
+            "application/json".parse::<ContentType>().unwrap(),
+            ContentType::json(),
+        );
+        println!("{:?}", "invalid/mimetypeeee".parse::<ContentType>());
+        assert!("invalid-mimetype".parse::<ContentType>().is_err());
```

Comment:
it would be nice to test which error shows up here, but `::Error` doesn't have an `Eq` and we also can't match on it because the `Kind` field is private, so I just went with this. I think it's ok.

It would be nice if we could produce a `headers::Error` that had a `source` from the mime error, but I didn't want to scope up this PR, so left it like this.